### PR TITLE
Integrate the parsing code

### PR DESCRIPTION
**Pull request by anto-deepsource**

### CODE REVIEW

File: src/parser/ParserFactory.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Factory to create a suitable parser object on the run.
+ */
+export default class ParserFactory {
+  /**
+   * Creates and returns a parser object based on the file extension.
+   * @param extension The extension of the file to be parsed
+   * @param options Custom options to be passed to the parser if any
+   * @returns a parser object
+   */
+  static getParser(extension: string, options: object) {
```
Comment (srijanpaul-deepsource) [inline src/parser/ParserFactory.ts]: We should change the type of parser options to `Record<string, any>`.
Or `Object`. (`object` is deprecated in TS)

File: src/parser/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Parses the code using a suitable parser. The parser to be used is decided based on the extension obtained from file path.
+ * @param filePath The path of file to parse
+ * @param code The code inside the file to parse
+ * @param options Custom options to be passed to the parser if any
+ * @returns the AST of the parsed code
+ */
+// TODO: Change 'object' type for 'options' arg to a proper interface (type) after deciding on the options input structure
+export default function parse(filePath: string, code: string, options: object = {}) {
+  // Get the file extension
```
Comment (srijanpaul-deepsource) [inline src/parser/index.ts]: Some of these comments are (IMO) redundant.
The code is clear and concise enough that the intention is conveyed by reading it.
I think we can get rid of these.
Opinion?

**Review by srijanpaul-deepsource — APPROVED** [review]

File: src/parser/ParserFactory.ts (hunk 1 above)
Comment (anto-deepsource) [reply src/parser/ParserFactory.ts]: Fixed

File: src/parser/index.ts (hunk 2 above)
Comment (anto-deepsource) [reply src/parser/index.ts]: Agreed. Removed.

**Review by srijanpaul-deepsource — APPROVED** [review]

**Pull request merged**